components/profile/shared/lists/agreementList/AgreementDetails.tsx

```diff
@@ -105,6 +106,8 @@ export const AgreementDetails: React.FC<{
 
     setLoadingChanges(true);
 
+    if (!user) throw new Error("User is not logged in");
+
     if (type == "Vipps") {
```

Comment:
I think it might be preferred to throw before mutating `loadingChanges`, else the view will be stuck in a loading state

```suggestion
    if (!user) throw new Error("User is not logged in");

    setLoadingChanges(true);
```